src/javaharness/javatests/arcs/android/impl/CrdtTest.java

```diff
@@ -140,62 +135,11 @@ public static void testRejectRemoveTooOld() {
-    assert set1.applyOperation(
-        new CollectionOperation<>(
-            CollectionOperation.Type.ADD, new Data("two"), VersionMap.of("me", 2), "me"));
-    CRDTCollection<Data> set2 = new CRDTCollection<>();
-    assert set2.applyOperation(
-        new CollectionOperation<>(
-            CollectionOperation.Type.ADD, new Data("three"), VersionMap.of("you", 1), "you"));
-    assert set2.applyOperation(
-        new CollectionOperation<>(
-            CollectionOperation.Type.ADD, new Data("one"), VersionMap.of("you", 2), "you"));
-    MergeResult<?> result1 = set1.merge(set2.getData());
-
-    CollectionData<Data> expectedSet1 = new CollectionData<>();
-    expectedSet1.values.put(
-        "one", new VersionedValue<>(new Data("one"), VersionMap.of("me", 1, "you", 2)));
-    expectedSet1.values.put("two", new VersionedValue<>(new Data("two"), VersionMap.of("me", 2)));
-    expectedSet1.values.put(
-        "three", new VersionedValue<>(new Data("three"), VersionMap.of("you", 1)));
-    expectedSet1.version = VersionMap.of("you", 2, "me", 2);
-    CollectionChange<Data> modelChange1 = (CollectionChange<Data>) result1.modelChange;
-    assert modelChange1.changeType == ChangeType.MODEL
-        : "modelChange1.changeType should be ChangeType.MODEL";
-    assert collectionDeepEquals(
-            (CollectionData<Data>) modelChange1.modelPostChange.get(), expectedSet1)
-        : "Unexpected merge model change (1).";
-    assert changeDeepEquals(modelChange1, (CollectionChange<Data>) result1.otherChange)
-        : "modelChange1 must be equal otherChange1";
-
-    // Test removes also work in merge.
-    set1.applyOperation(
-        new CollectionOperation<>(
-            CollectionOperation.Type.REMOVE,
-            new Data("one"),
-            VersionMap.of("me", 2, "you", 2),
-            "me"));
-
-    MergeResult<?> result2 = set1.merge(set2.getData());
-    CollectionData<Data> expectedSet2 = new CollectionData<>();
-    expectedSet2.values.put("two", new VersionedValue<>(new Data("two"), VersionMap.of("me", 2)));
-    expectedSet2.values.put(
-        "three", new VersionedValue<>(new Data("three"), VersionMap.of("you", 1)));
-    expectedSet2.version = VersionMap.of("you", 2, "me", 2);
-    CollectionChange<Data> modelChange2 = (CollectionChange<Data>) result2.modelChange;
-    assert modelChange2.changeType == ChangeType.MODEL
-        : "modelChange2.changeType should be ChangeType.MODEL";
-    assert collectionDeepEquals(
-            (CollectionData<Data>) modelChange2.modelPostChange.get(), expectedSet2)
-        : "Unexpected merge model change (2).";
-    assert changeDeepEquals(modelChange2, (CollectionChange<Data>) result2.otherChange)
-        : "modelChange2 must be equal otherChange2";
+  @Test
+  public void testMergeModels() {
+    // This test cannot be moved, because it heavily uses package private methods and data members.
+    // TODO: refactor?
```

Comment:
These tests should live in the same package as the class(es) they're testing (just swapping out java/ for javatests/). It's ok to test package-private methods/members.

This file should probably live in javatests/arcs/crdt/

Reply:
moved to the right package